### DISCUSSION TITLE
Clustermap center color around 0 and update legend bar

### DISCUSSION THREAD

**Discussion by GordianArnav** [DingWB/PyComplexHeatmap#71] [Q&A]
I have a heat map that have values for example say -4 to 2, I use the argument 'center = 0' to center the color bar around 0 so that 0 values appear white.

However the colorbar legend does not update (Range goes from -4 to 2, with white value not corresponding to zero)? Does anybody know how to solve this
```
cm = ClusterMapPlotter(data=cross_tab,
                       col_cluster=True, col_cluster_metric='correlation',
                       row_cluster=True,row_cluster_metric='correlation',
                       label='SignLogPval',row_dendrogram=True,col_dendrogram=col_dendogram,center=0,
                       show_rownames=True,show_colnames=True,verbose=0,legend_gap=5,
                     cmap='bwr',xticklabels_kws={'labelrotation':-90,'labelsize':xlabel_size},
                     yticklabels_kws={'labelsize':ylabel_size},annot = cross_tab_annot, norm=n1,row_names_side='left',row_dendrogram_size=10,
                     fmt=f'',annot_kws = {'weight':'bold'})
```

**Comment by DingWB** [comment]
I just pushed a commit; I passed `center` to legend_kws in [here](https://github.com/DingWB/PyComplexHeatmap/blob/a4036211715a813ff7976136434e521e642c2a23/PyComplexHeatmap/utils.py#L524).

Could you please help me test it? This will give you a general idea (using matplotlib.colors.CenteredNorm to normalize the camp). If it didn't work, you can also make some changes to the code and make a pull request.

**Reply by GordianArnav** [reply]
It changes the colorbar legend, however the actual heatmap isn't changing

**Reply by GordianArnav** [reply]
If i just pass the argument, outside of legend_kws, it works

**Reply by DingWB** [reply]
Yeah. @GordianArnav  Sure, it should be outside of `legend_kws,` just like you did for the first time.
Does everything look OK? I would appreciate it if you could help me do more tests.

**Reply by DingWB** [reply]
Could you please also show your code and figure here?
